### ISSUE TITLE
Transpose or Swap Axes of Table Visualization

### ISSUE THREAD

**Issue by pate** [metabase/metabase#2618]
When I group and sum some 2D data, Metabase seems to choose the x and y axes for me. How do I swap the table orientation? Even when swapping the two grouping keys, it still chooses for me.

*  I am using the latest Chrome browser.
*  My computer's OS is OSX.
*  I'm running Metabase locally from the JAR
*  My database is Postgres.
*  My Metabase version is v0.17.

**Comment by camsaul** [comment]
@tlrobinson is this a duplicate of #612  / #1445?

**Comment by salsakran** [comment]
Yes.

**Comment by camsaul** [comment]
Closing as duplicate. @pate, this feature is planned as part of our upcoming 0.19 release which should ship within the week.

**Comment by tlrobinson** [comment]
Not exactly. The current iteration of chart settings doesn't allow you to specify the columns in a pivot table. We still use the lower cardinality dimension for the column headers (we will probably want to fix #664 before changing that)

**Comment by salsakran** [comment]
@tlrobinson I'd like to decouple this from #664 

There are lots of times where I want to swap low cardinality dimensions being pivoted. Doesn't make sense to disallow that right now.

**Comment by salsakran** [comment]
maybe a warning if the proposed column dimension is huge?

**Comment by joevandyk** [comment]
There any progress on this?

**Comment by sudeep1801** [comment]
Hi - any progress on this - our company loves metabase for the charts, and this feature is really important so that i can represent comparison charts without axis getting transposed  based on cardinality.

**Comment by sunny-t** [comment]
I found a temporary way around this, if you rearrange your query output you can "trick" the pivot into arranging the data as required.

**Comment by clemzizer** [comment]
Any updates on this subject ?

**Comment by clemzizer** [comment]
@sunny-t do you mind explaining your trick ?

**Comment by clemzizer** [comment]
@sunny-t Thanks! Gonna have to figure how to make that work with a cumulative sum ;(

**Comment by sunny-t** [comment]
@clemzizer Well it worked for a simple count, for example:

SELECT count(*) , Date, Item
FROM "reporting"
GROUP BY Date, Item
ORDER BY Date asc

Would produce a pivot that had the count in the row title and date in the column title with item in the field value. 

SELECT Item, Date, count(*)
FROM "reporting"
GROUP BY Date, Item
ORDER BY Date asc 

Would then create a pivot with Item in the row title and date in the column title with count in the field value.

**Comment by clemzizer** [comment]
@sunny-t Thanks i'll have a look at it

**Comment by cameronr** [comment]
@sunny-t This workaround doesn't work for me. Here's what I'm seeing:

If you SELECT A, B, C for a pivoted table, the last element (e.g. C) will aways be used as the values for the pivot but the X axis will always be set to whichever of A or B has the lower cardinality. doing SELECT B, A, C won't produce a different table because metabase is picking automatically.

I'm very excited to see @tlrobinson's change! Hope that it gets in 31.
